12factor/twelvefactor.go

```diff
@@ -0,0 +1,102 @@
+// Package twelvefactor provides types to represents 12factor applications,
+// which are defined in http://12factor.net/
+package twelvefactor
+
+import (
+	"io"
+	"time"
+)
+
+// Manifest describes a 12factor application and it's processes.
+type Manifest struct {
+	App
+	Processes []Process
+}
+
+// App represents a 12factor application. We define an application as a
+// collection of processes that share a common environment and container image.
+type App struct {
+	// Unique identifier of the application.
+	ID string
+
+	// Name of the application.
+	Name string
+
+	// A string representing the version of this App.
+	Version string
+
+	// The container image for this app.
+	Image string
+
+	// The shared environment variables for the individual processes.
+	Env map[string]string
+}
+
+// Process represents an individual Process of an App, which defines the command
+// to run within the container image.
+type Process struct {
```

Comment:
One thing missing from this right now is port mappings. My thought is that, if you provide a host/container port mapping, then it's implementation specific as to what that does (e.g. with ECS backend, you'll probably get an ELB). How to keep this generic enough so that it can describe SSL/HTTP/TCP/etc across all implementations is challenging.